### PR TITLE
ADIOS2: Build for both MPICH and MPItrampoline

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by eschnett** [comment]
I'm not sure this built the right set of packages. It doesn't seem to have built separate packages for MPICH and MPItrampoline. I'm marking this as a draft. How do I correct this?

File: A/ADIOS2/build_tarballs.jl [hunk 1]
```diff
@@ -121,15 +132,16 @@ dependencies = [
...
     Dependency(PackageSpec(name="zfp_jll")),
 ]
 
+all_platforms, platform_dependencies = MPI.augment_platforms(platforms)
+append!(dependencies, platform_dependencies)
+
 # Build the tarballs, and possibly a `build.jl` as well.
 # GCC 4 is too old for Windows; it doesn't have <regex.h>
 # GCC 5 is too old for FreeBSD; it doesn't have `std::to_string`
 build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
```
Comment (vchuravy) [inline A/ADIOS2/build_tarballs.jl:146]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, all_platforms, products, dependencies;
```

Comment (eschnett) [reply A/ADIOS2/build_tarballs.jl:152]: Done.

**Comment by giordano** [comment]
This should be good to go now also with regards to MPIPreferences?  Or should we fix https://github.com/JuliaPackaging/Yggdrasil/blob/31285889f3a3078ef06fec4bdac920fc56496d02/platforms/mpi.jl#L11-L12 first?

**Comment by vchuravy** [comment]
We need to either wait for https://github.com/JuliaRegistries/RegistryCI.jl/pull/448, or @eschnett ping me after the merge and I will merge it manually into the Registry, 

I have done all my tests on LAMMPS_jll so I think this LGTM

**Comment by eschnett** [comment]
This might be ready now!

**Comment by giordano** [comment]
This is building also for OpenMPI, right?  So that

> except that we should add OpenMPI support to avoid https://github.com/JuliaPackaging/JLLWrappers.jl/issues/40 becoming a problem immediately.

shouldn't be a concern anymore.

**Comment by eschnett** [comment]
Yes, I added OpenMPI support. The respective tests failed initially, so I assume it's properly tested.

**Comment by giordano** [comment]
@vchuravy any objections against merging this now?  If not, I'll merge this tomorrow

**Pull request merged**